sdk/go/common/util/cmdutil/profile.go

```diff
@@ -52,16 +54,43 @@ func CloseProfiling(prefix string) error {
 	pprof.StopCPUProfile()
 	trace.Stop()
 
+	// get up-to-date statistics
+	return writeMemoryProfile(prefix)
+}
+
+func writeMemoryProfile(prefix string) error {
 	mem, err := os.Create(fmt.Sprintf("%s.%v.mem", prefix, os.Getpid()))
 	if err != nil {
 		return fmt.Errorf("could not create memory profile: %w", err)
 	}
 	defer contract.IgnoreClose(mem)
 
-	runtime.GC() // get up-to-date statistics
+	runtime.GC()
 	if err = pprof.Lookup("allocs").WriteTo(mem, 0); err != nil {
 		return fmt.Errorf("could not write memory profile: %w", err)
 	}
-
 	return nil
 }
+
+func memoryProfileWriteLoop(prefix string) {
+	// Every 5 seconds write a memory profile (in case we crash before we get a chance)
+	for i := 0; ; i++ {
+		// Sleep for a second
```

Comment:
This comment still doesn't look right.

Reply:
Sorry, I need to be a bit more careful and check my diffs